### PR TITLE
Feature/infiltration scene

### PR DESCRIPTION
**Pull request by DevilSky93**

### CODE REVIEW

File: Assets/Infiltration/Scripts/AI/Checks/CheckEnemyInFOVRange.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+namespace Infiltration
+{
+    public class CheckEnemyInFOVRange : Node
+    {
+        private readonly Transform _transform;
+        private const int PlayerLayerMask = 1 << 6;
+
+        private readonly SpriteRenderer _renderer;
+        private readonly float _fovRange;
+        public CheckEnemyInFOVRange(Transform transform, SpriteRenderer renderer, float fovRange)
```
Comment (samidalil) [inline Assets/Infiltration/Scripts/AI/Checks/CheckEnemyInFOVRange.cs:13]: Consider implementing a constructor without arguments and using public properties tagged with the ```ExposedInVisualEditor``` attribute (which will be added later to the codebase) with ```set``` or ```init``` accessors.

File: Assets/Infiltration/Scripts/AI/Checks/CheckEnemyInRange.cs [hunk 2]
```diff
@@ -0,0 +1,37 @@
+using BehaviorTree;
+using UnityEngine;
+
+namespace Infiltration
+{
+    public class CheckEnemyInRange : Node
+    {
+        private readonly Transform _transform;
+        private readonly float _attackRange;
+        public CheckEnemyInRange(Transform transform, float attackRange)
```
Comment (samidalil) [inline Assets/Infiltration/Scripts/AI/Checks/CheckEnemyInRange.cs:10]: Same thing about the constructor and properties.

File: Assets/Infiltration/Scripts/AI/Tasks/TaskAttackPlayer.cs [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+using UnityEngine;
+
+namespace Infiltration
+{
+    public class TaskAttackPlayer : Node
+    {
+        private readonly Transform _transform;
+        private readonly float _attackRange;
+
+        public TaskAttackPlayer(Transform transform, float attackRange)
```
Comment (samidalil) [inline Assets/Infiltration/Scripts/AI/Tasks/TaskAttackPlayer.cs:11]: Same thing about the constructor and properties.

File: Assets/Infiltration/Scripts/AI/Tasks/TaskGoTowardEnemy.cs [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+using UnityEngine;
+
+namespace Infiltration
+{
+    public class TaskGoTowardEnemy : Node
+    {
+        private readonly Transform _transform;
+        private readonly float _speed;
+
+        public TaskGoTowardEnemy(Transform transform, float speed)
```
Comment (samidalil) [inline Assets/Infiltration/Scripts/AI/Tasks/TaskGoTowardEnemy.cs:11]: Same thing about the constructor and properties.

File: Assets/Infiltration/Scripts/AI/Tasks/TaskPatrol.cs [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+        private readonly Transform[] _waypoints;
+
+        private int _currentWaypointIndex;
+
+        private const float WaitTime = 1f; // in seconds
+        private float _waitCounter;
+        private bool _waiting;
+
+        private readonly float _speed;
+        public TaskPatrol(Transform transform, Transform[] waypoints, float speed)
```
Comment (samidalil) [inline Assets/Infiltration/Scripts/AI/Tasks/TaskPatrol.cs:17]: Same thing about the constructor and properties.

File: Assets/Infiltration/Scripts/UIManager.cs [hunk 6]
```diff
@@ -0,0 +1,51 @@
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
+namespace Infiltration
+{
+    public class UIManager : MonoBehaviour
+    {
+        private bool _stateGame;
+        public static bool StateSet;
```
Comment (samidalil) [inline Assets/Infiltration/Scripts/UIManager.cs:9]: Consider using a ```GameManager``` to centralize game running state, scene loading and event dispatching if needed.

**Review by samidalil — CHANGES_REQUESTED** [review]

**Comment by samidalil** [comment]
[Scene 2 : Infiltration](https://trello.com/c/o8Kvf3JB/5-scene-2-infiltration)

File: Assets/General/BehaviorTree/BehaviorTreeAgent.cs [hunk 7]
```diff
@@ -13,6 +14,10 @@ private void Start()
 
         private void Update()
         {
+            if (UIManager.StateSet)
```
[the remark this answers is not in the document]
Comment (DevilSky93) [reply Assets/General/BehaviorTree/BehaviorTreeAgent.cs]: Add additional check with inverter node to stop the BT to run other sequences

File: Assets/Infiltration/Scripts/AI/Checks/CheckEnemyInFOVRange.cs (hunk 1 above)
Comment (DevilSky93) [reply Assets/Infiltration/Scripts/AI/Checks/CheckEnemyInFOVRange.cs:13]: Empty constructor already existing in `Node`, waiting for the `ExposedInVisualEditor `

**Review by samidalil — APPROVED** [review]

**Pull request merged**